src/test/java/cms/gov/madie/measure/services/TestCaseServiceTest.java

```diff
@@ -504,6 +522,328 @@ public void testUpdateTestCaseUpdatesLastModifiedFields() {
+    String modifiedJson = testCaseService.enforcePatientId(null, measure, testCase);
+    assertNull(modifiedJson);
+  }
+
+  @Test
+  public void testEnforcePatientIdNoEntry() {
+    String json = "{\"resourceType\": \"Bundle\", \"type\": \"collection\"}";
+    String modifiedJson = testCaseService.enforcePatientId(json, measure, testCase);
+    assertEquals(modifiedJson, json);
+  }
+
+  @Test
+  public void testEnforcePatientIdNoResource() {
+    String json =
+        "{\"resourceType\": \"Bundle\", \"type\": \"collection\", \n"
+            + "  \"entry\" : [ {\n"
+            + "    \"fullUrl\" : \"http://local/Patient/1\"\n"
+            + "  } ]             }";
+    String modifiedJson = testCaseService.enforcePatientId(json, measure, testCase);
+    assertEquals(modifiedJson, json);
+  }
+
+  @Test
+  public void testEnforcePatientIdNoResourceType() {
+    String json =
+        "{\"resourceType\": \"Bundle\", \"type\": \"collection\", \n"
+            + "  \"entry\" : [ {\n"
+            + "    \"fullUrl\" : \"http://local/Patient/1\",\n"
+            + "    \"resource\" : {\n"
+            + "      \"id\" : \"testUniqueId\"\n"
+            + "    }\n"
+            + "  } ]             }";
+    String modifiedJson = testCaseService.enforcePatientId(json, measure, testCase);
+    assertEquals(modifiedJson, json);
+  }
+
+  @Test
+  public void testEnforcePatientIdNoPatientResourceType() {
+    String json =
+        "{\"resourceType\": \"Bundle\", \"type\": \"collection\", \n"
+            + "  \"entry\" : [ {\n"
+            + "    \"fullUrl\" : \"http://local/Patient/1\",\n"
+            + "    \"resource\" : {\n"
+            + "      \"id\" : \"testUniqueId\",\n"
+            + "      \"resourceType\" : \"NOTPatient\"    \n"
+            + "    }\n"
+            + "  } ]             }";
+    String modifiedJson = testCaseService.enforcePatientId(json, measure, testCase);
+    assertEquals(modifiedJson, json);
+  }
+
+  @Test
+  public void testEnforcePatientIdNoTestCaseSeries() {
+    TestCase noSeriesTC = TestCase.builder().title("Test Title").build();
+    Measure testMeasure =
+        Measure.builder()
+            .cqlLibraryName("Test1CQLLibraryName")
+            .version(Version.builder().major(0).minor(0).revisionNumber(1).build())
+            .build();
+    testMeasure.setTestCases(List.of(noSeriesTC));
+    String json =
+        "{\"resourceType\": \"Bundle\", \"type\": \"collection\", \n"
+            + "  \"entry\" : [ {\n"
+            + "    \"fullUrl\" : \"http://local/Patient/1\",\n"
+            + "    \"resource\" : {\n"
+            + "      \"id\" : \"testUniqueId\",\n"
+            + "      \"resourceType\" : \"Patient\"    \n"
+            + "    }\n"
+            + "  } ]             }";
+    String modifiedJson = testCaseService.enforcePatientId(json, testMeasure, noSeriesTC);
+    assertNotEquals(modifiedJson, json);
+    assertTrue(modifiedJson.contains("TestTitle-Test1CQLLibraryName-0.0.1"));
+  }
+
```

Comment:
can we add a test case for malformed JSON? What happens in this case?

Reply:
@nmorasb testUpdateTestCaseEnforcingdPatientIdFail has an invalid json. Thanks Nate for reviewing the PR!